src/endpoint.rs

```diff
@@ -500,7 +517,18 @@ impl Endpoint {
             return Ok(());
         }
         self.connect_to(dest).await?;
-        self.try_send_message(msg, dest).await
+
+        let mut attempts: usize = 0;
+        let mut res = self.try_send_message(msg.clone(), dest).await;
+
+        while attempts < MAX_ATTEMPTS && res.is_err() {
```

Comment:
we will try resends for 10 times, then on disconnection, try re-connect for 10 times as well ?
and each takes an interval of 0.5s, which makes a total of 50s to detect a really failed connection ?

Reply:
Could be. (worth nothing, this has already been updated to `MAX_ATTEMPTS=5` ). 

What's an acceptable time here? How many tries do we want? 

50s doesn't seem too bad. _to be sure_, vs just a 30s timeout eg

Reply:
mainly just confirming this setup is ok.